routes/marketdata.js

```diff
@@ -13,48 +13,42 @@ const ref = db.ref("live_weller");
 // Links to markets list.
 const marketsRef = ref.child("markets");
 
-router.get('/:marketName', function(req, res, next) {
-    const marketName = req.params.marketName;
-
-    marketsRef.once('value', function(snapshot) {
-        snapshot.forEach(function(childSnapshot) {
-            const childData = childSnapshot.val();
-            
-            let status = "";
-            let questions = []
-            let missed = [];
-
-            if (childData.marketInfo.marketName === marketName) {
-                status = "Questions to fix to get to level " + (parseInt(childData.marketInfo.marketLevel) + 1);
+router.get('/:marketKey', function(req, res, next) {
+    const market = marketsRef.child(req.params.marketKey);
+
+    market.once('value', function(snapshot) {
+        let questions = []
```

Comment:
should move these down to their respective loops (right before the first time they're used) for clarity